CenterForOpenScience/ember-osf-web
# [ENG-1693][ENG-1704] Read only view for draft registration + Mute register button for write-users

- Ticket: [ENG-1704] [ENG-1693]
- Feature flag: n/a

## Purpose
- Add a new view for read-only contributors for draft registrations
- Disable Register button for non-admin users (code for this function stolen from [@brianjgeiger 's PR here](https://github.com/CenterForOpenScience/ember-osf-web/pull/1141)

## Summary of Changes
- Desktop: Only show link to `Review` page for read-only contributors on the left-hand side navigation
- Desktop: Remove `Back` button below the `Register` button on the right-hand side navigation
- Mobile: Remove hamburger menu that opens links to other pages on the top-bar
- Mobile: Remove back arrow that allows users to go to the previous page on the top-nav
- Desktop and Mobile: Disable register button and right nav for non-admin contributors
- Desktop and Mobile: Remove edit button in the Review page for read-only contributors
- Reroute read-only users to `Review` page when they enter a draft

## Screenshots
- Desktop, read-only
![Screen Shot 2021-02-16 at 9 50 22 AM](https://user-images.githubusercontent.com/51409893/108079087-66cd2080-703c-11eb-8951-85e3e7101096.png)

- Mobile, read-only
![Screen Shot 2021-02-16 at 9 51 34 AM](https://user-images.githubusercontent.com/51409893/108079208-91b77480-703c-11eb-98e1-fa1581050d4d.png)


## Side Effects
- None

## QA Notes
- Please make sure that read-only contributors are limited to the Review page of a draft registration
  - They should be routed to the `Review` page when they enter a draft and should be rerouted to the `Review` page even after manually trying to go to the `Metadata` page or any other page in the draft
  - They should not be able to click the Register button on the right-nav
  - They should not see the `Back` button on the right-nav
  - They should not be able to navigate away from the Review page using the left-nav or the right-nav buttons
  - They should not be able to see the edit buttons (pencil icons) on the Review page
- Please make sure that read-write users cannot register the draft
- Please make sure that only Admins are able to click the `Register` button on the Review page



[ENG-1704]: https://openscience.atlassian.net/browse/ENG-1704
[ENG-1693]: https://openscience.atlassian.net/browse/ENG-1693
